src/kakarot/instructions/system_operations.cairo

```diff
@@ -212,19 +212,31 @@ namespace SystemOperations {
         // 0 - size: byte size to copy
         // 1 - offset: byte offset in the memory in bytes
         let (stack, popped) = Stack.pop_n(self=stack, n=2);
-        // TODO: implement loading of the revert reason based on size value,
-        // currently limited by short string size
+
         let size = popped[0];
         let offset = popped[1];
 
         // Load revert reason from offset
-        let (memory, revert_reason_uint256, gas_cost) = Memory.load(memory, offset.low);
-        local revert_reason = revert_reason_uint256.low;
+        let (revert_reason_bytes: felt*) = alloc();
+        let (memory, gas_cost) = Memory.load_n(memory, size.low, revert_reason_bytes, offset.low);
 
         // revert with loaded revert reason short string
+        let truncate_reason = is_le(32, size.low);
```

Comment:
Is this less than or equal? In which case maybe there is a problem with the edge case 32?

Reply:
a felt can contains `31` bytes, so if the request size is `32` of greater, we need to truncate to `31`, right?